src/c++/perf_analyzer/genai-pa/genai_pa/parser.py

```diff
@@ -152,34 +168,42 @@ def add_profile_args(parser):
     )
 
 
-def add_endpoint_args(parser):
+def _add_endpoint_args(parser):
     endpoint_group = parser.add_argument_group("Endpoint")
 
+    endpoint_group.add_argument(
+        "-i",
+        type=str.lower,
+        choices=["http", "grpc"],
+        default="http",
```

Comment:
This can be a later decision if needed, but my 2cents from similar work is that using a default of GRPC+streaming will work out of box for **all triton models** when using the infer endpoint, so most users won't even need to think about setting this arg. If you wanted to use a decoupled model in triton on the infer endpoint (which seems like a sensible default for LLM tasks), then you **need** to use GRPC streaming anyway - so it would be better to not make the user have to set these args everytime for the "common case".

You can also consider selecting the endpoint default values based on the `--service-kind` wherever it applies.

ex something like this:
- `triton infer` -> default = grpc streaming
- `triton-generate` -> default = http
- `openai` -> default = http

Reply:
This is something i needed to put more thought into so I am glad you are bringing it up.
I think going based on service-kind is a good  method to handle this. Thank you for the suggestion

Reply:
I will handle this in the next PR is that is okay with you.